src/units/combine.rs

```diff
@@ -215,29 +232,38 @@ mod test {
             }
         ).unwrap();
 
-        // Set all units to stalled, check that the target goes stalled.
-        join!(u1.status(Stalled), u2.status(Stalled), u3.status(Stalled));
-        assert_eq!(t.recv().await, Err(Stalled));
+        // Set one unit to stalled, this triggers picking a new source healthy
+        // with data but there isn’t one, so we go stalled.
+        u1.send_stalled().await;
+        t.recv_stalled().await.unwrap();
+
+        // Now stall the other ones. That shouldn’t change anything.
+        // the last one.
+        u2.send_stalled().await;
+        t.recv_nothing().unwrap();
+        u3.send_stalled().await;
+        t.recv_nothing().unwrap();
 
-        // Set one unit to healthy.
-        u2.data(testrig::update(&[1])).await;
-        u2.status(Healthy).await;
-        assert_eq!(t.recv().await, Err(Healthy));
-        assert_eq!(t.recv().await, Ok(testrig::update(&[1])));
+        // Set one unit to healthy, check that we get an update.
```

Comment:
```suggestion
        // Set one unit to healthy by sending an update, which should be received by the target.
```

Reply:
Rephrased it in a slightly different way.